cypress/support/tooltip.po.ts

```diff
@@ -7,8 +7,112 @@ export class TooltipPo extends BaseComponent {
+    basic: 'demo-tooltip-basic',
+    placement: 'demo-tooltip-placement',
+    dismiss: 'demo-tooltip-dismiss',
+    dynamicTooltip: 'demo-tooltip-dynamic',
+    customContentTemplate: 'demo-tooltip-custom-content',
+    dynamicHtmlButton: 'demo-tooltip-dynamic-html',
+    defaultTooltip: 'demo-tooltip-container',
+    configuringDefaults: 'demo-tooltip-config',
+    customTriggersTooltip: 'demo-tooltip-triggers-custom',
+    manualTriggeringTooltip: 'demo-tooltip-triggers-manual',
+    componentLevelStyling: 'demo-tooltip-styling-local',
+    customClass: 'demo-tooltip-class',
+    delayTooltip: 'demo-tooltip-delay'
+
+  };
+
+  placementsOfTheTooltipContainer = {
+    placementTop: { tooltipContainerClass: 'tooltip in tooltip-top bs-tooltip-top top show', name: 'Tooltip on top' },
+    placementRight: {
+      tooltipContainerClass: 'tooltip in tooltip-right bs-tooltip-right right show',
+      name: 'Tooltip on right'
+    },
+    placementAuto: {
+      tooltipContainerClass: 'tooltip in tooltip-auto bs-tooltip-auto auto show bottom',
+      name: 'Tooltip auto'
+    },
+    placementLeft: {
+      tooltipContainerClass: 'tooltip in tooltip-left bs-tooltip-left left show',
+      name: 'Tooltip on left'
+    },
+    placementBottom: {
+      tooltipContainerClass: 'tooltip in tooltip-bottom bs-tooltip-bottom bottom show',
+      name: 'Tooltip on bottom'
+    }
   };
+
+  isTooltipPlacementCorrect(baseSelector: string, placement: string, buttonName: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`)
+      .contains(buttonName)
+      .focus();
+
+    cy.get(`${baseSelector} ${this.containerTooltip}`)
+      .should('be.visible')
+      .should('have.attr', 'class', placement);
+  }
+
+  isTooltipAppearOnFocus(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).focus();
+    this.isTooltipAppearsAndVisible(baseSelector);
+  }
+
+  isTooltipAppearOnClick(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).click();
+    this.isTooltipAppearsAndVisible(baseSelector);
+  }
+
+ isTooltipAppearsAndVisible(baseSelector: string) {
+    cy.get(`${baseSelector}`).should('to.have.descendants', this.containerTooltip);
+    cy.get(`${baseSelector} ${this.containerTooltip}`).should('be.visible');
+  }
+
+  isTooltipDismissAfterTheClicks(baseSelector: string) {
+    cy.get(`${baseSelector}`).closest('.section.bd-example').click();
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerTooltip);
+  }
+
+  isAppendToBodyTooltipAppearsVisibleAndDismissible(baseSelector: string, buttonName: string) {
+    cy.get(`${baseSelector}`).contains(buttonName).focus();
+    cy.get(`${this.containerTooltip}`).contains('Vivamus sagittis lacus vel augue laoreet rutrum faucibus');
+    this.isTooltipAppearsAndVisible(this.body);
+      }
+
+  isConfiguringTooltipAppearOnFocus(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).focus();
```

Comment:
this method not necessary, because you'll move focus to other method. In this case you'll use 2 methods directly in test